File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Wing.java
package com.nopalsoft.ponyrace.game_objects;

import java.util.Random;

public class Wing extends BaseGameObject {
    public float lastStateTime;
    public float stateTime;
    public State state;


    public Wing(float x, float y, Random oRan) {
        super(x, y);
        stateTime = oRan.nextFloat() * 5f;
        lastStateTime = stateTime;
        state = State.IDLE;
    }

    public void update(float delta) {
        lastStateTime = stateTime;
        stateTime += delta;
    }

    public enum State {
        IDLE, ACTIVE
    }
}

File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Wing.kt
package com.nopalsoft.ponyrace.game_objects

import java.util.Random

class Wing(x: Float, y: Float, oRan: Random) : BaseGameObject(x, y) {

    var stateTime = oRan.nextFloat() * 5F
    var lastStateTime = stateTime
    var state = State.IDLE

    fun update(delta: Float) {
        lastStateTime = stateTime
        stateTime += delta
    }

    enum class State {
        IDLE, ACTIVE
    }
}